src/main/java/com/mattworzala/canary/server/instance/block/LecternHandler.java

```diff
@@ -53,11 +58,30 @@ public boolean onInteract(@NotNull Interaction interaction) {
                 .append(Component.text(testFailure))
                 .build();
 
+        TextComponent errorText;
+        String hasteURL = testHasteUrl;
+        try {
+            if (testHasteUrl == null) {
+                Hastebin hastebin = new Hastebin();
+                hasteURL = hastebin.post(testStacktrace, false);
+                interaction.getInstance().setBlock(interaction.getBlockPosition(), block.withTag(Tags.TestHasteUrl, hasteURL));
+            }
+
+            TextComponent urlComponent = Component.text(hasteURL)
+                    .clickEvent(ClickEvent.openUrl(hasteURL));
+
+            errorText = Component.text("Full Stack trace: ").append(urlComponent);
+        } catch (IOException e) {
+            e.printStackTrace();
+
+            errorText = Component.text(testStacktrace);
+        }
+
         player.openBook(Book.builder()
                 .title(Component.text("Test Result"))
                 .author(Component.text("Canary Reporter"))
                 .addPage(simpleErrorPage)
-                .addPage(Component.text(testStacktrace))
+                .addPage(errorText == null ? Component.text(testStacktrace) : errorText)
```

Comment:
I wonder if this could just go on the first page and not have two at all? The stack trace page is useless anyway.